src/api/utils/marshmallow_schemas.py

```diff
@@ -182,20 +182,14 @@ class LogEditActivitySchema(BaseSchema):
         validate=[validate.Length(max=36)], load_from='activityId'
     )
     activity_type_id = fields.String(
-        validate=[validate.Length(max=36)], load_from='activityTypeId'
+        validate=[validate.Length(max=36)], load_from='activityTypeId', required=False
     )
     date = fields.Date()
     no_of_participants = fields.Integer(load_from='noOfParticipants')
 
     @validates_schema
     def validate_logged_activity(self, data):
         """Validate the logged activity."""
-        if (data.get('activity_type_id') and data.get('activity_id')) or not \
```

Comment:
I highly discourage removing this. This same Marshmallow class is used while creating a logged activity and as such, the code written was necessary to properly validate a `POST`. Alternatives you could consider are:
  - Inheriting from this class and overriding the `validate_logged_activity` method at line `196/7`
  - Use marshmallow's `context`. All Schema instances include a `context` attribute that would be useful when you want to disable this check during editing. You can find usage examples in the codebase or look [here](https://marshmallow.readthedocs.io/en/2.x-line/extending.html#using-context)

Reply:
I see this @LarryWachira.
Let me update.